### PR TITLE
Add an integration test for correlation logging

### PR DESCRIPTION
**Pull request by pasindu-sandima**
An integration test is added for the feature enable correlation
logs without a server restart for the apim product.

> Fix: wso2/api-manager#275
> Fix: wso2/api-manager#532

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+                        APIMIntegrationConstants.AM_KEY_MANAGER_INSTANCE, TestUserMode.SUPER_TENANT_ADMIN);
+        serverConfigurationManager = new ServerConfigurationManager(superTenantKeyManagerContext);
+    }
+
+    @Test(groups = {"wso2.am" }, description = "Testing the default correlation configs using the devops API ")
+    public void testRetrieveDefaultCorrelationLoggingConfigsTest() throws Exception {
+
+        //Retrieve default correlation logs configs from the GET method of the configs resource in devops API
+        HttpResponse loggingResponse =
+                HTTPSClientUtils.doGet(getStoreURLHttps() + "api/am/devops/v0/config/correlation", header);
```
Comment (SanojPunchihewa) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java:112]: Shall we define a constant for `api/am/devops/v0/config/correlation`?

**Comment by SanojPunchihewa** [comment]
Please remove unnecessary empty lines and fix the formatting

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+            "{\"name\":\"synapse\",\"enabled\":\"false\",\"properties\":[]}," +
+            "{\"name\":\"method-calls\",\"enabled\":\"false\",\"properties\":[]}]}";
+        Assert.assertEquals(loggingResponse.getData(),expectedResponse);
+
+        String logLine;
+        while ((logLine = bufferedReader.readLine()) != null) {
+            log.info("Start log : " + logLine);
+        }
+
+        // Create an application
```
Comment (SanojPunchihewa) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java:127]: Let's move the API and Application creation to `initialize()`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java (hunk 1 above)
Comment (pasindu-sandima) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java:112]: Sure

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+        }
+    }
+
+    private void InvokeTestAPI() throws Exception {
+        HttpClient client = HttpClientBuilder.create().setHostnameVerifier(new AllowAllHostnameVerifier()).build();
+        HttpGet request = new HttpGet(getAPIInvocationURLHttp(API_CONTEXT, API_VERSION));
+        request.setHeader("Authorization", "Bearer " + accessToken);
+        request.setHeader("activityid", "9e3ec6ed-2a37-4b20-8dd4-d5fbc754a7d9");
```
Comment (SanojPunchihewa) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java:287]: Use the constant `CORRELATION_ID` here as well

Comment (pasindu-sandima) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java:287]: Sure

**Comment by tharikaGitHub** [comment]
Closing as this PR is being addressed through https://github.com/wso2/product-apim/pull/13042

**Pull request closed without merging**